jgrapht-core/src/main/java/org/jgrapht/alg/interval/KorteMoehringIntervalGraphRecognizer.java

```diff
@@ -339,96 +324,175 @@ public boolean isIntervalGraph()
         // TODO implement
         return null;
     }
-    
-    
-    
-    
-    
-    
-    
-    
-    
-    
-    private abstract class MPQNode
-    {
-        MPQNode left;
-        MPQNode right;
-        MPQNode parent;
-        
-        MPQNodeSetElement bag;
-        
-        MPQNode(MPQNodeSetElement bag) {
-            this.bag = bag;
+
+    /**
+     * A node of a modified PQ-tree
+     */
+    private abstract class MPQTreeNode {
+
+        /**
+         * The parent of the current node
+         */
+        MPQTreeNode parent;
+
+        /**
+         * Both P node and Q node must keep track of the left most and right most child
+         */
+        MPQTreeNode leftmostChild;
+        MPQTreeNode rightmostChild;
+
+        /**
+         * The graph vertices associated with the current tree node
+         *
+         * The associated set of vertices is given by a doubly linked circular list
+         */
+        List<V> elements = null; // TODO: replace with doubly linked circular list on the LHS
+
+        /**
+         * The label of the current tree node
+         */
+        Label label;
+
+        /**
+         * Instantiate a tree node associating with no graph vertex
+         */
+        MPQTreeNode() { }
+
+        /**
+         * Instantiate a tree node associating with a set of graph vertices
+         * TODO: replace this with an addElement method later
+         *
+         * @param elements a set of graph vertices associated with this tree node
+         */
+        MPQTreeNode(List<V> elements) {
+            this.elements = elements;
         }
-        
-        abstract void add(MPQNode newChild);
+
+        /**
+         * Add child to the current tree node
+         *
+         * @param newChild the child node to be added
+         */
+        abstract void addChild(MPQTreeNode newChild);
```

Comment:
I actually want to remove this, because add Child is only reasonable for PNodes.